lib/src/test/kotlin/app/cash/kfsm/Letter.kt

```diff
@@ -5,7 +5,11 @@ data class Letter(override val state: Char) : Value<Letter, Char> {
   override fun update(newState: Char): Letter = copy(state = newState)
 }
 
-sealed class Char(to: () -> Set<Char>) : State(to)
+sealed class Char(to: () -> Set<Char>) : State<Char>(to) {
+  fun next(count: Int): List<Char> =
+    if (count <= 0) emptyList()
+    else subsequentStates.filterNot { it == this }.firstOrNull()?.let { listOf(it) + it.next(count - 1) } ?: emptyList()
```

Comment:
Without this change, you would not be able to recursively call `next` here (or call it anywhere else in a callback for example)

Reply:
The real test is that this compiles, but there is [a test](https://github.com/cashapp/kfsm/pull/48/files#diff-acbdd798bb3eb1abbb67aaf9ce0d8feb353213b8afc6842d7b622cecd1405f88R58-R61) that uses this method also just to prove it out fully.